superset/views/core.py

```diff
@@ -2091,6 +2111,35 @@ def favstar(self, class_name, obj_id, action):
         session.commit()
         return json_success(json.dumps({'count': count}))
 
+    @api
```

Comment:
Can we move this to a separate `.py` file. `core.py` is getting really big and should start trying to breakdown the endpoints into their separate file. 

@mistercrunch Can i get your opinion on this?

Reply:
I totally agree, `core.py` is kind of becoming a monolith which is less than ideal. In the same song I'm not sure if this is the feature to start breaking things out, it seems like it should be a whole refactoring effort so that concerns can be separated and it can be clearer what the PR is aiming at. For instance there's a `dashboard.py` file adjacent but that seems to only contain the logic for creating a new dashboard at this moment. I do think that everything with the name `Dashboard` should eventually move into that file (`DashboardModelView`, `DashboardAddView`, `DashboardFilter`, etc) but that seems like one rather self-contained task to me.

Reply:
We recently started breaking apart the large modules into packages. In its current state, `core.py` stands for "views that haven't been sorted out yet".

The main challenge related to this is that the permission associated to the view need to be migrated properly. Anyhow, new dashboard-related views should be in `views/dashboard.py`, but we'll need an effort to move other views later regardless. 

File under "optional" for now